locator/tablets.cc

```diff
@@ -285,13 +347,40 @@ class tablet_effective_replication_map : public effective_replication_map {
         if (!info) {
             return {};
         }
-        tablet_logger.trace("get_pending_endpoints({}): table={}, tablet={}, replica={}",
-                            search_token, _table, tablet, info->pending_replica);
-        return {get_endpoint_for_host_id(info->pending_replica.host)};
+        switch (info->writes) {
+            case write_replica_set_selector::previous:
+                return {};
+            case write_replica_set_selector::both:
+                tablet_logger.trace("get_pending_endpoints({}): table={}, tablet={}, replica={}",
+                                    search_token, _table, tablet, info->pending_replica);
+                return {get_endpoint_for_host_id(info->pending_replica.host)};
+            case write_replica_set_selector::next:
+                return {};
+        }
```

Comment:
We should pre-calculate anything that changes only during edges, like info->writes.

Reply:
You mean avoid the switch here? In this case, it saves only the switch. There's not much computation going on.

Reply:
We could change it to return a span\<> later. But okay for now, since we'd need to copy into the return value anyway.